### ISSUE TITLE
Conditions in defVar, addConstraint indexing

### ISSUE THREAD

**Issue by joehuchette** [JuliaOpt/JuMP.jl#161]
Contrived example:
```
@defVar(m, x[i=1:10,j=1:10;i+j==8])
```
Sudoku example (set initial solutions):
```
@addConstraint(m, inits[row=1:9,col=1:9,v=initgrid[i,j];v!=0], x[row, col, v] == 1)
```

**Comment by mlubin** [comment]
Is this feasible now given the parser changes in 0.4?

**Comment by joehuchette** [comment]
I think so?
```jl
julia> dump(:(x[i=1:3;isodd(i)]))
Expr
  head: Symbol typed_vcat
  args: Array(Any,(3,))
    1: Symbol x
    2: Expr
      head: Symbol =
      args: Array(Any,(2,))
        1: Symbol i
        2: Expr
          head: Symbol :
          args: Array(Any,(2,))
          typ: Any
      typ: Any
    3: Expr
      head: Symbol call
      args: Array(Any,(2,))
        1: Symbol isodd
        2: Symbol i
      typ: Any
  typ: Any

julia> dump(:(x[i=1:3,j=1:3;isodd(i)]))
Expr
  head: Symbol typed_vcat
  args: Array(Any,(4,))
    1: Symbol x
    2: Expr
      head: Symbol parameters
      args: Array(Any,(1,))
        1: Expr
          head: Symbol call
          args: Array(Any,(2,))
          typ: Any
      typ: Any
    3: Expr
      head: Symbol =
      args: Array(Any,(2,))
        1: Symbol i
        2: Expr
          head: Symbol :
          args: Array(Any,(2,))
          typ: Any
      typ: Any
    4: Expr
      head: Symbol =
      args: Array(Any,(2,))
        1: Symbol j
        2: Expr
          head: Symbol :
          args: Array(Any,(2,))
          typ: Any
      typ: Any
  typ: Any
```

**Comment by mlubin** [comment]
Closed by #277

**Comment by yeesian** [comment]
:+1:
